dataflux_pytorch/benchmark/checkpointing/simulated/multi_node_benchmark.py

```diff
@@ -49,56 +44,66 @@ def init_processes() -> int:
     os.environ["NODE_RANK"] = str(rank)
 
     configure_master_addr()
-    torch.distributed.init_process_group(
-        backend='gloo', rank=rank, world_size=world_size)
+    torch.distributed.init_process_group(backend='gloo',
+                                         rank=rank,
+                                         world_size=world_size)
     return rank
 
 
-def main(world_size: int, model_size: int, project: str, filepath: str, padding_size: int, sample_size: int) -> None:
+def main(world_size: int, layer_size: int, project: str, filepath: str,
+         padding_size: int, sample_count: int) -> None:
     rank = init_processes() if os.environ.get("COORDINATOR_ADDRESS") else 0
-    model = SimpleModel(model_size, padding_size)
-
-    dummy_input = torch.randn(100, model_size)
-    _ = model(dummy_input)
-
-    full_state_dict = model.state_dict()
-    for i, tensor in enumerate(model.dummy_tensors):
-        full_state_dict[f'dummy_tensor_{i}'] = tensor
 
     benchmark_strategy = BenchmarkStrategy(
-        project=project, path=filepath, model=model)
-
-    distributed_state_dict = {f"{key}_shard_{rank}": split_tensor(
-        tensor, world_size, rank) for key, tensor in full_state_dict.items()}
-
-    save_checkpoint_times = time_checkpoint_operation(
-        benchmark_strategy, distributed_state_dict, filepath, sample_size, 'save')
-    load_checkpoint_times = time_checkpoint_operation(
-        benchmark_strategy, distributed_state_dict, filepath, sample_size, 'load')
+        project=project,
+        path=filepath,
+    )
+
+    state_dict = dict()
+    for i in range(padding_size):
+        if i % world_size == rank:
+            state_dict[f'dummy_tensor_{i}'] = torch.randn(layer_size, 1000)
+    # Wait until the state_dict is populated properly accross all the nodes.
+    dist.barrier()
+
+    save_checkpoint_times = time_checkpoint_operation(benchmark_strategy,
+                                                      state_dict, filepath,
+                                                      sample_count, 'save',
+                                                      rank, world_size,
+                                                      padding_size, layer_size)
+
+    load_checkpoint_times = time_checkpoint_operation(benchmark_strategy,
+                                                      state_dict, filepath,
+                                                      sample_count, 'load',
+                                                      rank, world_size,
+                                                      padding_size, layer_size)
 
     if rank == 0:
-        print("######################")
```

Comment:
Did you mean to get rid of this print at the start?

Reply:
Nope I just feel it helps capture the entire block between two set of ###